src/ecalc/libraries/libecalc/common/libecalc/input/mappers/create_references.py

```diff
@@ -56,12 +56,58 @@ def create_references(configuration: PyYamlYamlModel, resources: Resources) -> R
         fuel_data.get(EcalcYamlKeywords.name): FuelMapper.from_yaml_to_dto(fuel_data)
         for fuel_data in configuration.fuel_types
     }
+
+    consumers_installations = []
+    for installation in configuration.installations:
+        if installation.get(EcalcYamlKeywords.generator_sets) is not None:
+            consumers_installations.append(
+                [
+                    consumer
+                    for consumers in installation[EcalcYamlKeywords.generator_sets]
+                    for consumer in consumers[EcalcYamlKeywords.consumers]
+                ]
+            )
+
+        if installation.get(EcalcYamlKeywords.fuel_consumers) is not None:
+            consumers_installations.append(list(installation[EcalcYamlKeywords.fuel_consumers]))
+
+    check_multiple_energy_models(consumers_installations)
+
     return References(
         models=models,
         fuel_types=fuel_types,
     )
 
 
+def check_multiple_energy_models(consumers_installations: list):
+    """
+    Check for different energy model types within one consumer.
+
+    Args:
+        consumers_installations (list): List of consumers per installation
+
+    Returns:
+        Value error if different energy model types found within one consumer
```

Comment:
Is it more correct to say it "returns None"
```suggestion
     Raises:
        Value error if different energy model types found within one consumer
```

Reply:
I agree, returns None is more reasonable. Will fix.